src/test/unit_tests/test_s3_bucket_policy.js

```diff
@@ -1963,4 +1968,129 @@ mocha.describe('s3_bucket_policy', function() {
             assert.strictEqual(res.PolicyStatus.IsPublic, true);
         });
     });
+    mocha.describe('s3_bucket_policy: Granting access to a specific version of an object ', function() {
```

Comment:
I think you need to split the `decribe` into small test cases, it is hard to follow.
You can do the preparing in the `describe section` and then separate it into different cases.
I understand that not all actions are tested, but if we want to add more cases, it will be easier.

Reply:
I agree with you. However, I would not modify it at this point. I have been testing these tests for sometime and looks like well cooked. Will do it when I have to add more tests in this in future.